File: src/main/java/com/martmists/ynet/network/Channel.java
package com.martmists.ynet.network;

import com.martmists.ynet.api.BaseProvider;

import java.util.HashSet;
import java.util.Set;

public class Channel {
    public Class<? extends BaseProvider> providerType;
    public Set<ConnectorConfiguration> connectorSettings;

    public Channel() {
        providerType = BaseProvider.class;
        connectorSettings = new HashSet<>();
    }
}

File: src/main/kotlin/com/martmists/ynet/network/Channel.kt
package com.martmists.ynet.network

import com.martmists.ynet.transfer.type.DisabledType
import com.martmists.ynet.transfer.type.Type

class Channel<T : Type>(val network: Network, val connectedBlocks: MutableList<ConfiguredBlockEntity>, var type: T) {
    companion object {
        fun empty(network: Network) = Channel(network, mutableListOf(), DisabledType)
    }
}
